main/IpsDisplay.cpp

```diff
@@ -945,24 +945,37 @@ void IpsDisplay::drawCable(int16_t x, int16_t y)
 {
 	const int16_t CANH = 8;
 	const int16_t CANW = 14;
-	CAN->connectedXCV() ? ucg->setColor(COLOR_LBLUE) : ucg->setColor(COLOR_MGREY);
+	bool CANconnectedXCV = false;
+	bool CANconnectedMag = false;
+	if (CAN) {
+		CANconnectedXCV = CAN->connectedXCV();
+		CANconnectedMag = CAN->connectedMagSens();
+	}
+	if (CANconnectedXCV)
+		ucg->setColor(COLOR_LBLUE);
+	else
+		ucg->setColor(COLOR_MGREY);
+	if (CANconnectedMag)
+		ucg->setColor(COLOR_GREEN);
 	// ucg->setFont(ucg_font_fub11_hr);
 	// ucg->setPrintPos(x - 8, y);
-	if (CAN->connectedMagSens()) {
-		ucg->setColor(COLOR_GREEN);
-	}
 	ucg->drawLine( x-CANW/2, y+CANH/2, x+3, y+CANH/2 );
 	ucg->drawLine( x-CANW/2, y+CANH/2-1, x+3, y+CANH/2-1 );
 	ucg->drawDisc( x-CANW/2, y+CANH/2, 2, UCG_DRAW_ALL);
 	// ucg->print("c");
-	CAN->connectedMagSens() ? ucg->setColor(COLOR_LBLUE) : ucg->setColor(COLOR_MGREY);
-	if (Flarm::connected()) {
+	if (CANconnectedMag)
+		ucg->setColor(COLOR_LBLUE);
+	else
+		ucg->setColor(COLOR_MGREY);
+	if (Flarm::connected())
 		ucg->setColor(COLOR_GREEN);
-	}
 	ucg->drawLine( x+2, y+CANH/2, x-4, y-CANH/2 );
 	ucg->drawLine( x+3, y+CANH/2-1, x-3, y-CANH/2-1 );
 	// ucg->print("a");
-	CAN->connectedXCV() ? ucg->setColor(COLOR_LBLUE) : ucg->setColor(COLOR_MGREY);
```

Comment:
That would work too, they are equivalent, although I found this use of (condition? do_A : do_B) a little hard to read.  I am used to using it for assignment: C=(condition? A : B) but here it is part of the execution flow which is unusual.

Reply:
We get more confusion if we change the syntax just because of readability. The cond ? A : B; syntax is a longterm feature in C++. Just change it in your branch and it should appear here.